src/carousel.tsx

```diff
@@ -0,0 +1,699 @@
+    carouselContainer: {
+        position: 'relative',
+    },
+    arrowButton: {
+        ...arrowButtonStyle(theme),
+        position: 'absolute',
+        zIndex: 1,
+        top: `calc(50% - ${arrowButtonSize / 2}px)`,
+        '&.prev': {
+            left: -arrowButtonSize / 2,
+            [theme.mq.tabletOrSmaller]: {
+                left: ({sideMargin}) => -sideMargin,
+            },
+            [theme.mq.largeDesktop]: {
+                left: -(24 + arrowButtonSize),
+            },
+        },
+        '&.next': {
+            right: -arrowButtonSize / 2,
+            [theme.mq.tabletOrSmaller]: {
+                right: ({sideMargin}) => -sideMargin,
+            },
+            [theme.mq.largeDesktop]: {
+                right: -(24 + arrowButtonSize),
+            },
+        },
+    },
+    carousel: {
+        display: 'flex',
+        overflowX: 'auto',
+        minWidth: '100%',
+        scrollSnapType: ({free}) => (free ? 'initial' : 'x mandatory'),
+        ...hideScrollbar,
+        [theme.mq.tabletOrSmaller]: {
+            margin: ({sideMargin}) => `0 -${sideMargin}px`,
+            '&::before, &::after': {
+                content: '""',
+                flexShrink: 0,
+                display: 'block',
+                width: ({centered}) => (centered ? '25%' : 0),
+            },
+        },
+    },
+    slide: {
+        scrollSnapAlign: 'start',
+        flexShrink: 0,
+        width: ({itemsPerPageConfig, gap}) =>
+            `calc(${100 / itemsPerPageConfig.desktop}% + ${gap / itemsPerPageConfig.desktop}px)`,
+        scrollMargin: ({gap}) => `-${gap}px`,
+        '&:first-child': {
+            width: ({itemsPerPageConfig, gap}) =>
+                `calc(${100 / itemsPerPageConfig.desktop}% - ${
+                    (gap * (itemsPerPageConfig.desktop - 1)) / itemsPerPageConfig.desktop
+                }px)`,
+            scrollMargin: 0,
+        },
+        [theme.mq.tabletOrSmaller]: {
+            scrollSnapAlign: ({centered}) => (centered ? 'center' : 'start'),
+            scrollMargin: ({mobilePageOffsetConfig}) => `${mobilePageOffsetConfig.prev}px`,
+            width: ({itemsPerPageConfig, mobilePageOffsetConfig, gap}) =>
+                `calc(${100 / itemsPerPageConfig.mobile}% - ${
+                    (mobilePageOffsetConfig.next + mobilePageOffsetConfig.prev + gap) /
+                    itemsPerPageConfig.mobile
+                }px)`,
+            '&:first-child': {
+                paddingLeft: ({centered, sideMargin}) => (centered ? 0 : sideMargin),
+                width: ({itemsPerPageConfig, mobilePageOffsetConfig, gap, centered, sideMargin}) =>
+                    `calc(${100 / itemsPerPageConfig.mobile}% - ${
+                        (mobilePageOffsetConfig.next + mobilePageOffsetConfig.prev + gap) /
+                            itemsPerPageConfig.mobile +
+                        gap -
+                        (centered ? 0 : sideMargin)
+                    }px)`,
```

Comment:
perhaps you could add a `/* prettier-ignore */` comment here and format manually